app/controllers/api/v1/clinics_controller.rb

```diff
@@ -0,0 +1,45 @@
+class Api::V1::ClinicsController < ApplicationController
+  before_action :set_clinic, only: %i[show update destroy]
+
+  def index
+    @clinics = Clinic.all
+    render json: @clinics
+  end
+
+  def show
+    render json: @clinic
+  end
+
+  def create
+    @clinic = Clinic.new(clinic_params)
+    @clinic.doctor = Doctor.find(params[:doctor_id])
+
+    if @clinic.save
+      render json: @clinic, status: :created
+    else
+      render json: @clinic.errors, status: :unprocessable_entity
+    end
+  end
+
+  def update
+    if @clinic.update(clinic_params)
+      render json: @clinic
+    else
+      render json: @clinic.errors, status: :unprocessable_entity
+    end
+  end
+
+  def destroy
+    @clinic.destroy
+  end
```

Comment:
- [ ] What happens if the clinic is not destroyed correctly and the request fails? Kindly add some error handling here. 👍